cpp/src/arrow/builder.h

```diff
@@ -913,6 +914,21 @@ Status ARROW_EXPORT MakeDictionaryBuilder(MemoryPool* pool,
                                           const std::shared_ptr<DataType>& type,
                                           std::shared_ptr<ArrayBuilder>* out);
 
+/// \brief Encodes a given Array using a hash table
+///
+/// \param[in] input The Array to be encoded
+/// \param[in] pool MemoryPool to allocate memory for the hash table
+/// \param[out] out Array encoded to dictionaries
+Status ARROW_EXPORT EncodeDictionary(const Array& input, MemoryPool* pool,
+                                     std::shared_ptr<Array>* out);
+
+/// \brief Encodes a given ChunkedArray using a hash table
+///
+/// \param[in] input The ChunkedArray to be encoded
+/// \param[in] pool MemoryPool to allocate memory for the hash table
+/// \param[out] out ChunkedArray encoded to dictionaries
```

Comment:
These comments aren't quite accurate. The input and output are Column, and if I'm reading correctly, the function concatenates all of the arrays rather than preserving the existing chunk layout